realm/realm-library/src/androidTest/java/io/realm/RealmTests.java

```diff
@@ -3332,4 +3334,24 @@ public void run(Realm realm) {
         TestHelper.awaitOrFail(bgRealmFished);
         assertFalse(bgRealmChangeResult.get());
     }
+
+    @Test
+    public void incompatibleLockFile() throws IOException {
+        // Replace .lock file with a corrupted one
+        File lockFile = new File(realmConfig.getPath() + ".lock");
+        assertTrue(lockFile.exists());
+        FileOutputStream fooStream = new FileOutputStream(lockFile, false);
+        fooStream.write("Boom".getBytes());
+        fooStream.close();
```

Comment:
maybe add a `flush` to ensure the file is modified before proceeding with the rest of the test